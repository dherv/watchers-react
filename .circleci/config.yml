version: 2
jobs:
  build:
    working_directory: ~/tmp
    branches:
      only:
        - develop
        - feature/*
        - master
    docker:
      - image: circleci/node
    steps:
      - checkout
      - run: yarn install
      - run: yarn test
