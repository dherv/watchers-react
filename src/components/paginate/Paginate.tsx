import React, { SFC } from "react";

interface IProps {
  count: number;
  current_page: number;
  next_page: number;
  previous_page: number;
  onClick: (page: number) => void;
}
const Paginate: SFC<IProps> = ({
  count,
  current_page,
  next_page,
  previous_page,
  onClick
}) => {
  const number_of_page: number[] = Array(Math.ceil(count / 23));
  // added "downlevelIteration": true to original create-react-app tsconfig.json to allow [...number_of_page.keys()]
  const iterator = [...number_of_page.keys()].map((key: number) => key + 1);

  const handleClick = (page: number) => {
    onClick(page);
  };
  return (
    <ul>
      {iterator.map((page: number) => (
        <li key={page} onClick={() => handleClick(page)}>
          {page}
        </li>
      ))}
    </ul>
  );
};

export default Paginate;
