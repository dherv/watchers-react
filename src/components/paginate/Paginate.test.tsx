import { renderer, shallow, React } from "../../test.config";
import Paginate from "./Paginate";
describe("Paginate", () => {
  const props = {
    count: 120,
    current_page: 2,
    next_page: 3,
    previous_page: 1,
    onClick: jest.fn()
  };
  const number_of_pages = Math.ceil(props.count / 23);
  const wrapper = shallow(<Paginate {...props} />);
  describe("component", () => {
    it("should display one <ul>", () => {
      expect.assertions(1);
      expect(wrapper.find("ul")).toHaveLength(1);
    });
    it("should display a number of <li> equal to number_of_pages", () => {
      expect.assertions(1);
      expect(wrapper.find("li")).toHaveLength(number_of_pages);
    });
    it("should call prop onClick when <li> is clicked", () => {
      expect.assertions(1);
      const li = wrapper.find("li").first();
      li.simulate("click");
      expect(props.onClick).toHaveBeenCalledTimes(1);
    });
  });
  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const tree = renderer.create(<Paginate {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
