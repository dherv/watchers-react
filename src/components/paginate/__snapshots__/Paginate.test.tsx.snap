// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginate snapshot should match 1`] = `
<ul>
  <li
    onClick={[Function]}
  >
    1
  </li>
  <li
    onClick={[Function]}
  >
    2
  </li>
  <li
    onClick={[Function]}
  >
    3
  </li>
  <li
    onClick={[Function]}
  >
    4
  </li>
  <li
    onClick={[Function]}
  >
    5
  </li>
  <li
    onClick={[Function]}
  >
    6
  </li>
</ul>
`;
